src/HotChocolate/Core/src/Types/Types/Descriptors/Conventions/DefaultNamingConventions.cs

```diff
@@ -332,17 +334,67 @@ public virtual bool IsDeprecated(object value, out string? reason)
     }
 
     /// <inheritdoc />
-    public NameString FormatFieldName(string fieldName)
+    public unsafe NameString FormatFieldName(string name)
```

Comment:
@michaelstaib  does this work for you? 
This is still a single iteration over the name but split into two loops. I do have an implementation with just one `for` but it actually is a lot more comprehesible this way